code/week5/animal.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <iostream>
+
+class Animal {
```

Comment:
Base classes in C++ which you intend to be used polymorphically must declare virtual destructors (even if they are just default). All classes here which are derived from should have a `~T() = default;` line in their declaration.